### PR TITLE
Fix Axios issue when using http proxy agent

### PR DESCRIPTION
**Pull request by attiasas**
Fixing Axios issue: https://github.com/axios/axios/issues/4531

Should resolve the issues:
* https://github.com/jfrog/jfrog-vscode-extension/issues/247
* https://github.com/jfrog/jfrog-vscode-extension/issues/97

### CODE REVIEW

File: src/HttpClient.ts [hunk 1]
```diff
@@ -70,6 +72,20 @@ export class HttpClient {
...
+    ): HttpsProxyAgent | undefined {
+        if (
+            !proxyConfig ||
+            !proxyConfig.host ||
+            !proxyConfig.port ||
+            (proxyConfig.protocol && proxyConfig.protocol.includes('https'))
+        ) {
+            return undefined;
+        }
+        return new HttpsProxyAgent(`http://${proxyConfig.host}:${proxyConfig.port}`);
```
Comment (yahavi) [inline src/HttpClient.ts:86]: What happens if it is an https->https proxy?

File: src/HttpClient.ts [hunk 2]
```diff
@@ -70,6 +72,20 @@ export class HttpClient {
...
     }
 
+    public static getHttpToHttpsProxyConfig(
+        proxyConfig: IProxyConfig | false | undefined
+    ): HttpsProxyAgent | undefined {
+        if (
+            !proxyConfig ||
+            !proxyConfig.host ||
+            !proxyConfig.port ||
+            (proxyConfig.protocol && proxyConfig.protocol.includes('https'))
```
Comment (yahavi) [inline src/HttpClient.ts:82]: This check repeats here and in getAxiosProxyConfig

File: src/HttpClient.ts (hunk 1 above)
Comment (attiasas) [reply src/HttpClient.ts:86]: can't be, we checked just before if the proxy protocol is https return undefined

File: src/HttpClient.ts (hunk 2 above)
Comment (attiasas) [reply src/HttpClient.ts:82]: not exactly, here we check if the protocol **is** 'https' and return undefined if they do (i.e use default)
at getAxiosProxyConfig we check if the protocol **is not** 'https', i.e getHttpToHttpsProxyConfig will not return undefined, therefore we need to return false in order to disable the default proxy handling

File: test/tests/HttpClient.spec.ts [hunk 3]
```diff
@@ -23,6 +24,46 @@ describe('Http client tests', () => {
...
+            config: false,
+            shouldExists: false,
+        },
+    ].forEach((testCase) => {
+        test('Constructing HttpToHttpsProxyConfig - ' + testCase.name, () => {
+            let res: HttpsProxyAgent | undefined = HttpClient.getHttpToHttpsProxyConfig(
+                <IProxyConfig | false | undefined>testCase.config
+            );
+            if (testCase.shouldExists) {
+                expect(res).toBeDefined();
```
[the remark this answers is not in the document]
Comment (attiasas) [reply test/tests/HttpClient.spec.ts:60]: done

File: src/HttpClient.ts [hunk 4]
```diff
@@ -70,6 +72,20 @@ export class HttpClient {
         }
     }
 
+    public static getHttpToHttpsProxyConfig(
```
[the remark this answers is not in the document]
Comment (attiasas) [reply src/HttpClient.ts:75]: done

**Pull request merged**